src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,41 @@
+package duke;
+
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Dealine Class to represent a class that will store Deadline object that extends from Tasks Class
+ * @author amresh A0235398R
+ */
+public class Deadline extends Task {
+    protected String by;
+    protected LocalDate date;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Method to convert deadline String to LocalDate format
+     * @return boolean true value if converted
+     */
+    public boolean convertToDate() {
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+            LocalDate localDate = LocalDate.parse(this.by, formatter);
+            this.date = localDate;
+            this.by = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
```

Comment:
Perhaps you could avoid using `this` here since the variable names are not being shadowed?
```suggestion
            date = localDate;
            by = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
```